### PR TITLE
refactor: cleanup `NetworkIcon`

### LINKED ISSUE #7146: [Task]: Cleanup `NetworkIcon`

### PR DESCRIPTION
**Pull request by evavirseda**
## Summary

Closes #7146 

## Changelog

```
- cleanup and add missing types
```

### Platforms

> Please select any platforms where your changes have been tested.

-   **Desktop**
    -   [ ] MacOS
    -   [ ] Linux
    -   [ ] Windows
-   **Mobile**
    -   [ ] iOS
    -   [ ] Android

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to **test and verify** that your changes work as intended.

...

## Checklist

> Please tick the following boxes that are relevant to your changes.

-   [ ] I have followed the contribution guidelines for this project
-   [ ] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have added or modified tests that prove my changes work as intended
-   [ ] I have verified that new and existing unit tests pass locally with my changes
-   [ ] I have verified that my latest changes pass CI workflows for testing and linting
-   [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/shared/components/atoms/NetworkIcon.svelte [hunk 1]
```diff
@@ -4,38 +4,55 @@
...
 </script>
 
-<network-icon
-    class="flex items-center justify-center p-0.5 rounded-full bg-{backgroundColor} {outlined
-        ? 'ring-2 ring-white dark:ring-gray-900'
-        : ''}"
->
+<network-icon class={`${outlined ? 'outlined' : ''} ${getBackgroundClasses(backgroundColor)}`}>
     <Icon {height} {width} icon={NETWORK_ICON_SVG[networkId]} classes="text-{iconColor}" />
 </network-icon>
```
Comment (VmMad) [inline packages/shared/components/atoms/NetworkIcon.svelte:48]: I think this component can be refactored a bit more. The enum and the function are unnecessary. You can make a map and use the networkId to index it. 

```svelte
// ...
    $: backgroundColor = classesMap[networkId]?.backgroundColor ?? ''
    $: iconColor = classesMap[networkId]?.iconColor ?? ''

    const classesMap: { [key in NetworkId]?: Record<string, string> } = {
        [NetworkId.Iota]: {
            backgroundColor: 'bg-black',
            iconColor: 'text-white',
        },
        [NetworkId.Shimmer]: {
            backgroundColor: 'bg-shimmer-highlight',
            iconColor: 'text-black',
        },
        [NetworkId.Testnet]: {
            backgroundColor: 'bg-gray-400',
            iconColor: 'text-black',
        },
    }
</script>

<network-icon class={backgroundColor ? backgroundColor : null} class:outlined>
    <Icon {height} {width} icon={NETWORK_ICON_SVG[networkId]} classes={iconColor} />
</network-icon>
```

File: packages/shared/components/NetworkIcon.svelte [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            iconColor: 'text-black',
+        },
+        [NetworkId.Testnet]: {
+            backgroundColor: 'bg-gray-400',
+            iconColor: 'text-black',
+        },
+    }
+</script>
+
+<network-icon class={backgroundColor ? backgroundColor : null} class:outlined>
```
Comment (VmMad) [inline packages/shared/components/NetworkIcon.svelte:30]:
```suggestion
<network-icon class={backgroundColor ?? null} class:outlined>
```

Comment (VmMad) [reply packages/shared/components/NetworkIcon.svelte:30]: or even better just change the reactive line `$: backgroundColor = classesMap[networkId]?.backgroundColor ?? null` so you can leave `class={backgroundColor}`

File: packages/shared/components/NetworkIcon.svelte [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        },
+        [NetworkId.Testnet]: {
+            backgroundColor: 'bg-gray-400',
+            iconColor: 'text-black',
+        },
+    }
+</script>
+
+<network-icon class={backgroundColor} class:outlined>
+    <Icon {height} {width} icon={NETWORK_ICON_SVG[networkId]} classes="text-{iconColor}" />
```
Comment (VmMad) [inline packages/shared/components/NetworkIcon.svelte:31]: you forgot :thinking:  

```suggestion
    <Icon {height} {width} icon={NETWORK_ICON_SVG[networkId]} classes={iconColor} />
```

**Review by VmMad — CHANGES_REQUESTED** [review]

**Review by VmMad — APPROVED** [review]
Nice and clean :broom:

**Review by cpl121 — APPROVED** [review]
Nice job! 🥇

**Pull request merged**